lib/components/Atom/WheelPicker/index.tsx

```diff
@@ -0,0 +1,231 @@
+            );
+            const selectedElement = Math.min(
+                Math.max(Math.floor(scrollTop / 48), 0),
+                data.length - 1,
+            );
+            (dataRefs.current[selectedElement] as HTMLDivElement)?.focus({
+                preventScroll: true,
+            });
+            setSelectedValue(data[selectedElement].value);
+            isAnimating = false;
+        }
+    };
+
+    const resizeObserver = new ResizeObserver(() => {
+        const index = Number(currentDataValue.current);
+        if (!dataItemsContRef?.current) return;
+        const divHeight = dataItemsContRef?.current.clientHeight;
+        dataItemsContRef.current.style.paddingTop = `${divHeight * 0.5}px`;
+        dataItemsContRef.current.style.paddingBottom = `${divHeight * 0.5}px`;
+        if (dataRefs.current[index]) {
+            (dataRefs.current[index] as HTMLDivElement).scrollIntoView({
+                block: "center",
+            });
+        }
+    });
+
+    useEffect(() => {
+        if (!dataItemsContRef.current) return;
+        setSelectedValue(data[currentDataValue.current].value);
+        const divHeight = dataItemsContRef?.current.clientHeight;
+        resizeObserver.observe(dataItemsContRef.current);
+        dataItemsContRef.current.style.paddingTop = `${divHeight * 0.5}px`;
+        dataItemsContRef.current.style.paddingBottom = `${divHeight * 0.5}px`;
+        dataItemsContRef.current.addEventListener("scroll", handleDataScroll);
+        return () => {
+            dataItemsContRef.current?.removeEventListener(
+                "scroll",
+                handleDataScroll,
+            );
+        };
+    }, []);
+
+    useEffect(() => {
+        if (focus) {
+            dataRefs.current[dataItemsMap.get(selectedValue)]?.focus();
+        }
+    }, [focus]);
+
+    return (
+        <div
+            className={clsx(
+                "quill-wheel-picker__container",
+                containerClassName,
+            )}
+        >
+            <ul
+                className={clsx(
+                    "quill-wheel-picker__data-items",
+                    listClassName,
+                )}
+                role="listbox"
+                ref={dataItemsContRef}
+            >
+                {data.map(({ value }, index) => {
+                    return (
+                        <DropdownItem
+                            {...rest}
+                            className={clsx(
+                                "quill-wheel-picker__data-item",
+                                dropDownItemClassName,
+                            )}
+                            label={value}
+                            role="option"
+                            tabIndex={0}
+                            textAlignment="center"
+                            ref={(node: HTMLDivElement) =>
+                                (dataRefs.current[index] = node)
+                            }
+                            key={index}
```

Comment:
Is there any error regarding index as keys? Maybe we can use value instead? Please, ignore if irrelevant

Reply:
yes it has to be value